### PR TITLE
frontend/feat: #218 Multiple Merchant Support in Native Side

### LINKED ISSUE #218: Add Multiple Merchant Support
- Adding support for multiple merchants.
- Adding build variant in native side for different merchant.
- Adding config for different merchant.

### PR DESCRIPTION
**Pull request by Vignesh-772**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?

closes #218 

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/values/styles.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<resources>
+
+    <!-- Base application theme. -->
+    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
+        <!-- Customize your theme here. -->
+        <item name="colorPrimary">@color/colorPrimary</item>
+        <item name="colorPrimaryDark">@color/colorAccent</item>
+        <item name="colorAccent">@color/colorAccent</item>
+<!--        <item name="materialAlertDialogTheme">@style/MaterialAppTheme</item>-->
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/values/styles.xml:9]: Can you remove these commented lines

File: Frontend/android-native/app/src/user/jatriSaathi/java/in/juspay/mobility/JsInterface.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            headerObj.put("x-client-id", "jatrisaathi");
+            boolean b = setAnalyticsHeader(headerObj.toString());
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+//        polylines = new ArrayList<>();
+
+    }
+
+//     @JavascriptInterface
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/user/jatriSaathi/java/in/juspay/mobility/JsInterface.java:99]: Can you remove there commented lines

File: Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_floating_widget.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_floating_widget.png:1]: 4 same images are there with the different names, Can you remove those

File: Frontend/android-native/app/src/user/nammaYatri/res/drawable/button.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+<!--        <item android:state_focused="true" android:state_pressed="false" android:drawable="@drawable/YOURIMAGE" />-->
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/user/nammaYatri/res/drawable/button.xml:3]: Can you remove commented lines

**Review by Rohit4997 — COMMENTED** [review]
Added coments, kindly check

**Review by ks-shree — CHANGES_REQUESTED** [review]
Can you add yatri merchant support also

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 4]
```diff
@@ -351,10 +349,7 @@ public void onFailure(@NonNull Exception e) {
         webView.getSettings().setJavaScriptEnabled(true);
         webView.getSettings().setLoadWithOverviewMode(true);
         webView.getSettings().setUseWideViewPort(true);
-        if (key != null && key.equals("nammayatripartner")) {
-            getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
-            CommonJsInterface.updateLocaleResource(sharedPref.getString(getResources().getString(R.string.LANGUAGE_KEY), "null"));
-        } else {
+        if (key != null && key.equals("nammayatri")) {
```
Comment (ks-shree) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:352]: @Vignesh-772 Added check if driver or customer app

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 5]
```diff
@@ -622,20 +621,6 @@ private void initApp() {
         } catch (JSONException e) {
             e.printStackTrace();
         }
-//        try {
```
Comment (ks-shree) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:625]: @Vignesh-772 Remove commented code

File: Frontend/android-native/app/src/driver/nammaYatriPartner/js/juspay/index_bundle.js
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/js/juspay/index_bundle.js:1]: @Vignesh-772 Remove default index_bundle.js

File: Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_driver_vehicle.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_driver_vehicle.png:1]: reduce the image size (3.44MB is a lot)

File: Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/index_bundle.js
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/index_bundle.js:1]: @Vignesh-772 Remove default index_bundle.js

Comment (ks-shree) [reply Frontend/android-native/app/src/driver/nammaYatriPartner/assets/juspay/index_bundle.js:1]: Same in other places also

File: Frontend/android-native/spagobundling.sh [hunk 6]
```diff
@@ -1,22 +1,33 @@
...
-mkdir -p "app/src/driver/js/juspay"
-rm -rf app/src/driver/js/juspay/index_bundle.js
-cp atlas-ui-driver/dist/index_bundle.js app/src/driver/js/juspay
+mkdir -p "app/src/driver/$2/js/juspay"
+rm -rf app/src/driver/$2/js/juspay/index_bundle.js
+cp ../ui-driver/dist/index_bundle.js app/src/driver/$2/js/juspay
+rm -rf app/src/driver/$2/assets/juspay/juspay_assets.json
+sh driverJuspayAssets.sh $2
+
 echo " ---------- CreateJSAFiles :- --------------"
```
Comment (ks-shree) [inline Frontend/android-native/spagobundling.sh:29]: @Vignesh-772 Remove createJSA commands

File: Frontend/.DS_Store
Comment (ks-shree) [inline Frontend/.DS_Store:1]: Delete /.DS_Store file , this is not supposed to be pushed

File: Frontend/.psc-ide-port
Comment (ks-shree) [inline Frontend/.psc-ide-port:1]: Delete this file

File: Frontend/android-native/app/ny_Prod/debug/output-metadata.json
Comment (ks-shree) [inline Frontend/android-native/app/ny_Prod/debug/output-metadata.json:1]: @Vignesh-772 Remove this file , It is not supposed to be pushed

File: Frontend/android-native/app/ny_p_Prod/debug/output-metadata.json
Comment (ks-shree) [inline Frontend/android-native/app/ny_p_Prod/debug/output-metadata.json:1]: Remove this file , It is not supposed to be pushed

File: Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/config.json [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+                "config": "https://assets.juspay.in/juspay/payments/in.juspay.upiintent/release/v1-config.zip"
+            }
+        },
+        "id": "48215399535214850",
+        "package": {
+            "in.juspay.ec": "https://assets.juspay.in/juspay/payments/in.juspay.ec/release/2.0rc1/v1-index_bundle.zip",
+            "in.juspay.godel": "https://assets.juspay.in/juspay/payments/in.juspay.godel/release/2.0rc1/v1-index_bundle.zip",
+            "in.juspay.godel.placeholder": "https://assets.juspay.in/juspay/payments/in.juspay.godel/release/2.0rc1/v1-index_bundle.zip",
+            "in.juspay.hyperpay": "https://assets.juspay.in/juspay/payments/in.juspay.hyperpay/release/2.0rc1/v1-index_bundle.zip",
+            "in.juspay.jatrisaathidriver": "https://assets.juspay.in/hyper/bundles/app/in.juspay.jatrisaathidriver/1.1.0/android/v1-index_bundle.zip",
```
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/config.json:106]: @Vignesh-772 For jatrisaathidriver bundle push is setted up?

File: Frontend/android-native/app/src/driver/yatriPartner/assets/juspay/config.json
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/yatriPartner/assets/juspay/config.json:1]: @Vignesh-772 Is this lastest config.json for Yatri app?

File: Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/config.json (hunk 7 above)
Comment (Vignesh-772) [reply Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/config.json:106]: not yet

File: Frontend/android-native/app/src/driver/nammaYatriPartner/js/juspay/index_bundle.js
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/js/juspay/index_bundle.js:1]: @Vignesh-772 Remove index_bundle.js present in both assets and js path

File: Frontend/android-native/app/src/driver/jatriSaathiDriver/res/raw/finding_rides_loader_with_text_v2.json
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/jatriSaathiDriver/res/raw/finding_rides_loader_with_text_v2.json:1]: @Vignesh-772 Why do we have finding_rides lottie json in driver ?

Can you remove if not required?

File: Frontend/android-native/app/src/driver/jatriSaathiDriver/res/raw/finding_rides_loader_with_text.json
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/jatriSaathiDriver/res/raw/finding_rides_loader_with_text.json:1]: Remove if not used

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/raw/finding_rides_loader_with_text.json
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/raw/finding_rides_loader_with_text.json:1]: @Vignesh-772 Why do we have finding_rides lottie json in driver ?

Can you remove if not required?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/raw/finding_rides_loader_with_text_v2.json
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/raw/finding_rides_loader_with_text_v2.json:1]: remove if not used

File: Frontend/android-native/app/src/driver/yatriPartner/res/raw/finding_rides_loader_with_text.json
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/yatriPartner/res/raw/finding_rides_loader_with_text.json:1]: @Vignesh-772 Why do we have finding_rides lottie json in driver ?

Can you remove if not required?

File: Frontend/android-native/app/src/driver/yatriPartner/res/raw/finding_rides_loader_with_text_v2.json
Comment (ks-shree) [inline Frontend/android-native/app/src/driver/yatriPartner/res/raw/finding_rides_loader_with_text_v2.json:1]: Check this also

File: Frontend/android-native/app/src/user/jatriSaathi/assets/juspay/juspay_assets.json
Comment (ks-shree) [inline Frontend/android-native/app/src/user/jatriSaathi/assets/juspay/juspay_assets.json:1]: Remove

File: Frontend/android-native/app/src/user/yatri/assets/juspay/config.json
Comment (ks-shree) [inline Frontend/android-native/app/src/user/yatri/assets/juspay/config.json:1]: @Vignesh-772 Add latest config.json

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 8]
```diff
@@ -54,9 +54,183 @@ view push state =
...
+  ][ (if os == "IOS" then linearLayout else scrollView)
+      [ height MATCH_PARENT
+      , width MATCH_PARENT
+      , scrollBarY false
+      ][ linearLayout
+          [ orientation VERTICAL
+          , height WRAP_CONTENT
+          , width MATCH_PARENT
+          , padding $ PaddingBottom 30
+          , margin $ MarginVertical 14 0
```
Comment (ks-shree) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:74]: This can be `MarginTop`

**Comment by MercyQueen** [comment]
Not required

**Pull request closed without merging**